qemu/exec.c

```diff
@@ -1484,14 +1489,17 @@ static void tcg_commit(MemoryListener *listener)
 
 void address_space_init_dispatch(AddressSpace *as)
 {
-    as->dispatch = NULL;
-    as->dispatch_listener = (MemoryListener) {
-        .begin = mem_begin,
-        .commit = mem_commit,
-        .region_add = mem_add,
-        .region_nop = mem_add,
-        .priority = 0,
+    MemoryListener ml = {
+        mem_begin,
+        mem_commit,
+        mem_add,
+        NULL,
+        mem_add,
+        NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,
+        0,
     };
+    as->dispatch = NULL;
+    as->dispatch_listener = ml;
     memory_listener_register(as->uc, &as->dispatch_listener, as);
```

Comment:
Yes the problem is that const structs cannot be initialised like this.
So instead of doing things in multiple different ways I just decided to choose the most generic way for all cases.

For some cases I did create C style constructors that would setup structs from member values, but again this only works for non-const ones.